### PR TITLE
Fix Jenkinsfile failure

### PR DESCRIPTION
**Pull request by causten**
I started seeing this failure when I merged #2201 .  Trying to figure out how to resolve 
[2023-09-28T19:28:14.222Z] WorkflowScript: 110: illegal string body character after dollar sign;
[2023-09-28T19:28:14.222Z]    solution: either escape a literal dollar sign "\$5" or bracket the value expression "${5}" @ line 110, column 160.
[2023-09-28T19:28:14.222Z]    E_C_FLAGS_DEBUG='${debug_flags}' -DMIGRA

### CODE REVIEW

**Review by TedThemistokleous — APPROVED** [review]

**Review by umangyadav — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -129,13 +128,13 @@ rocmtest clang_debug: rocmnode('cdna') { cmake_build ->
...
             def debug_flags = "-g -O2 -fsanitize=${sanitizers} -fno-sanitize=vptr -fno-sanitize-recover=${sanitizers}"
-            cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DMIGRAPHX_ENABLE_MLIR=On -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags_cxx}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}' -DGPU_TARGETS=$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')")
+            cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DMIGRAPHX_ENABLE_MLIR=On -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags_cxx}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}' -DGPU_TARGETS=\"$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')\"")
         }
     }
 }, ck_release: rocmnode('mi100+') { cmake_build ->
     stage('CK Release') {
         withEnv(['MIGRAPHX_ENABLE_CK=1', 'MIGRAPHX_TUNE_CK=1']) {
-            cmake_build(flags: "-DCMAKE_BUILD_TYPE=release -DGPU_TARGETS=$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')")
+            cmake_build(flags: "-DCMAKE_BUILD_TYPE=release -DGPU_TARGETS=\"$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')\"")
```
Comment (umangyadav) [inline Jenkinsfile:137]:
```suggestion
            cmake_build(flags: "-DCMAKE_BUILD_TYPE=release -DGPU_TARGETS='$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')'")
```

File: Jenkinsfile [hunk 2]
```diff
@@ -129,13 +128,13 @@ rocmtest clang_debug: rocmnode('cdna') { cmake_build ->
             // Note: the -fno-sanitize= is copied from upstream LLVM_UBSAN_FLAGS.
             def debug_flags_cxx = "-g -O2 -fsanitize=${sanitizers} -fno-sanitize=vptr,function -fno-sanitize-recover=${sanitizers}"
             def debug_flags = "-g -O2 -fsanitize=${sanitizers} -fno-sanitize=vptr -fno-sanitize-recover=${sanitizers}"
-            cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DMIGRAPHX_ENABLE_MLIR=On -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags_cxx}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}' -DGPU_TARGETS=$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')")
+            cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DMIGRAPHX_ENABLE_MLIR=On -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags_cxx}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}' -DGPU_TARGETS=\"$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')\"")
```
Comment (umangyadav) [inline Jenkinsfile:131]:
```suggestion
            cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DMIGRAPHX_ENABLE_MLIR=On -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags_cxx}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}' -DGPU_TARGETS='$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')'")
```

File: Jenkinsfile [hunk 3]
```diff
@@ -120,7 +119,7 @@ rocmtest clang_debug: rocmnode('cdna') { cmake_build ->
 //     }
 }, all_targets_debug : rocmnode('cdna') { cmake_build ->
     stage('All targets Release') {
-        cmake_build(flags: "-DCMAKE_BUILD_TYPE=release -DMIGRAPHX_ENABLE_GPU=On -DMIGRAPHX_ENABLE_CPU=On -DMIGRAPHX_ENABLE_FPGA=On -DGPU_TARGETS=$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')") 
+        cmake_build(flags: "-DCMAKE_BUILD_TYPE=release -DMIGRAPHX_ENABLE_GPU=On -DMIGRAPHX_ENABLE_CPU=On -DMIGRAPHX_ENABLE_FPGA=On -DGPU_TARGETS=\"$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')\"")
```
Comment (umangyadav) [inline Jenkinsfile:122]:
```suggestion
        cmake_build(flags: "-DCMAKE_BUILD_TYPE=release -DMIGRAPHX_ENABLE_GPU=On -DMIGRAPHX_ENABLE_CPU=On -DMIGRAPHX_ENABLE_FPGA=On -DGPU_TARGETS='$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')'") 
```

File: Jenkinsfile [hunk 4]
```diff
@@ -107,7 +106,7 @@ rocmtest clang_debug: rocmnode('cdna') { cmake_build ->
     stage('hipRTC Debug') {
         def sanitizers = "undefined"
         def debug_flags = "-g -O2 -fsanitize=${sanitizers} -fno-sanitize-recover=${sanitizers}"
-        cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}' -DMIGRAPHX_USE_HIPRTC=On -DGPU_TARGETS=$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')", gpu_debug: true)
+        cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}' -DMIGRAPHX_USE_HIPRTC=On -DGPU_TARGETS=\"$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')\"", gpu_debug: true)
```
Comment (umangyadav) [inline Jenkinsfile:109]:
```suggestion
        cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}' -DMIGRAPHX_USE_HIPRTC=On -DGPU_TARGETS='$(/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*')'", gpu_debug: true)
```

File: Jenkinsfile [hunk 5]
```diff
@@ -1,3 +1,7 @@
+def getgputargets() {
+    targets="gfx908;gfx90a;gfx1030;gfx1100;gfx1101;gfx1102"
```
Comment (umangyadav) [inline Jenkinsfile:2]: Need to add `gfx906` . Also grepping for `gfx` arch didn't work ?

**Pull request merged**